framework/src/meshmodifiers/RenameBlock.C

```diff
@@ -0,0 +1,87 @@
+  params.addParam<std::vector<SubdomainName> >("new_block_name", "Elements with the old block number (or name) will be given this block name.  You must supply either new_block_id or new_block_name.  You may supply a vector of new_block_id, in which case the old_block information must also be a vector.");
+  params.addClassDescription("RenameBlock re-numbers or re-names an old_block_id or old_block_name with a new_block_id or new_block_name");
+  return params;
+}
+
+RenameBlock::RenameBlock(const std::string & name, InputParameters parameters) :
+    MeshModifier(name, parameters)
+{
+  // error checking.  Must have exactly one of old_block_id or old_block_name
+  if (isParamValid("old_block_id") && isParamValid("old_block_name"))
+    mooseError("RenameBlock: You must supply exactly one of old_block_id or old_block_name\n");
+  else if (!isParamValid("old_block_id") && !isParamValid("old_block_name"))
+    mooseError("RenameBlock: You must supply exactly one of old_block_id or old_block_name\n");
+
+  // error checking.  Must have exactly one of new_block_id or new_block_name
+  // In principal we could have both (the old block would then be given a new ID and a new name)
+  // but i feel that could lead to confusion for the user.  If the user wants to do that they
+  // should use two of these RenameBlock MeshModifiers.
+  if (isParamValid("new_block_id") && isParamValid("new_block_name"))
+    mooseError("RenameBlock: You must supply exactly one of new_block_id or new_block_name\n");
+  else if (!isParamValid("new_block_id") && !isParamValid("new_block_name"))
+    mooseError("RenameBlock: You must supply exactly one of new_block_id or new_block_name\n");
+}
+
+RenameBlock::~RenameBlock()
+{
+}
+
+void
+RenameBlock::modify()
+{
+  // Check that we have access to the mesh
+  if (!_mesh_ptr)
+    mooseError("_mesh_ptr must be initialized before calling RenameBlock::modify()");
+
+  // grab the user input.  Can't do all this in the constructor as some things may not
+  // have been put into the mesh yet, eg old_block_name might have been inserted by
+  // another MeshModifier
+  if (isParamValid("old_block_id"))
+    _old_block_id = getParam<std::vector<SubdomainID> >("old_block_id");
+  else // must have supplied old_block_name
+    _old_block_id = _mesh_ptr->getSubdomainIDs(getParam<std::vector<SubdomainName> >("old_block_name"));
+
+
+  if (isParamValid("new_block_id"))
+  {
+    _new_block_id = getParam<std::vector<SubdomainID> >("new_block_id");
+    if (_new_block_id.size() != _old_block_id.size())
+      mooseError("RenameBlock: The vector of old_block information must have the same length as the vector of new_block information\n");
+    for (unsigned i = 0 ; i < _old_block_id.size() ; ++i)
+      for (MeshBase::element_iterator el = _mesh_ptr->getMesh().active_elements_begin(); el != _mesh_ptr->getMesh().active_elements_end(); ++el)
```

Comment:
Not a big deal, but you might reverse the order of these two loops so you only do one loop over elements.